### DISCUSSION TITLE
What's the difference between the kind of a tracing span?

### DISCUSSION THREAD

**Discussion by adinhodovic** [linkerd/linkerd2#4979] [General] [answered]
![image](https://user-images.githubusercontent.com/12065867/93397457-ee3e1a00-f879-11ea-9c3d-3d067e718b01.png)

What's the difference in the above example between the two spans? And what does `kind` mean in the above example? There's two spans registered by Linkerd-proxy for each request and the only difference is the kind.

**Comment by ihcsim** [comment, marked as answer]
A typical request flow looks like this: ----> inbound server ----> inbound client ---> application ----> outbound server ----> outbound client --->. (Thanks @adleong for the explanation.) The `span.kind` and `direction` labels tell us which side the span is from. If you are interested, there is more info on how the proxy handles requests in this blog [post](https://linkerd.io/2020/07/23/under-the-hood-of-linkerds-state-of-the-art-rust-proxy-linkerd2-proxy/#the-life-of-a-request).

**Reply by adinhodovic** [reply]
Thanks a lot, makes sense! Will read.
